src/SiteMaster/Core/Auditor/Site/Page/ScanForm.php

```diff
@@ -166,7 +165,21 @@ public function handlePost($get, $post, $files)
      */
     protected function scan($get, $post, $files)
     {
-        $page_scan = Page::createNewPage($this->scan->id, $this->site->id, $this->uri, $this->page->found_with, array(
+        if (!$this->page && $this->scan->getDistinctPageCount() >= Config::get('SCAN_PAGE_LIMIT')) {
```

Comment:
This business logic seems like it would be better suited as a method of the scan. Not sure how many places use it.